src/entities/auth/model/authStore.ts

```diff
@@ -0,0 +1,45 @@
+import { create } from "zustand";
+import { persist } from "zustand/middleware";
+
+interface User {
+  email: string;
+  nickname?: string;
+}
+
+interface AuthStore {
+  isAuthenticated: boolean;
+  accessToken: string | null;
+  refreshToken: string | null;
+  user: User | null;
+  setUser: (user: User) => void;
+  setTokens: (accessToken: string, refreshToken: string) => void;
+  setAccessToken: (accessToken: string) => void;
+  clearAuth: () => void;
+}
+
+export const useAuthStore = create<AuthStore>()(
+  persist(
+    (set) => ({
+      isAuthenticated: false,
+      accessToken: null,
+      refreshToken: null,
+      user: null,
+      setUser: (user) => set({ user }),
+      setTokens: (accessToken, refreshToken) => set({ accessToken, refreshToken, isAuthenticated: true }),
+      setAccessToken: (accessToken) => set({ accessToken }),
+      clearAuth: () => set({ accessToken: null, refreshToken: null, user: null, isAuthenticated: false }),
+    }),
```

Comment:
로그인 여부는 accessToken 여부로 판단할 수 있으니 상태를 각각 두지 말고 아래처럼 accessToken 기반으로 자동 계산되도록 하면 좋을 것 같아요! onRehydreateStorage 도 삭제

```suggestion
export const useAuthStore = create<AuthStore>()(
  persist(
    (set, get) => ({
      isAuthenticated: () => !!get().accessToken,
      accessToken: null,
      refreshToken: null,
      user: null,
      setUser: (user) => set({ user }),
      setTokens: (accessToken, refreshToken) => set({ accessToken, refreshToken }),
      setAccessToken: (accessToken) => set({ accessToken }),
      clearAuth: () => set({ accessToken: null, refreshToken: null, user: null }),
    }),
```